fmf/_compat_jsonschema.py

```diff
@@ -0,0 +1,91 @@
+"""Compatibility layer for jsonschema validation."""
+
+from __future__ import annotations
+
+import warnings
+from typing import Any, Dict, List, NamedTuple, Optional, Union
+
+import jsonschema
+
+try:
+    from jsonschema import validators
+    from referencing import Registry, create
+    from referencing.jsonschema import DRAFT4
+    MODERN_JSONSCHEMA = True
+except ImportError:
+    MODERN_JSONSCHEMA = False
```

Comment:
Could we make the test be on the `importlib` package version instead? It is less clear what `MODERN_JSONSCHEMA` is referring to otherwise. As for the `create_validator`, what about adding the imports in there?

Reply:
How about now, without the flag?

Reply:
👍